porcupine/tabs.py

```diff
@@ -687,8 +688,31 @@ def reload(self, *, undoable: bool = True) -> bool:
         assert self.textwidget["state"] == "normal"
         self.textwidget.config(state="disabled")
 
+        def confirm_encoding(encoding: str) -> str | None:
+            return utils.ask_encoding(
+                f'The content of "{self.path}" is not valid '
+                + self.settings.get("encoding", str)
+                + ". Choose an encoding to use instead:",
+                encoding,
+            )
+
         while True:
             try:
+                if self.settings.get("encoding", str) == "utf-8":
+                    with self.path.open("rb") as fb:
+                        byte_str = fb.read(4)
+                    for bytez, guessed_encoding in [
+                        (codecs.BOM_UTF8, "utf-8-sig"),
+                        (codecs.BOM_UTF32_LE, "utf-32"),
+                        (codecs.BOM_UTF32_BE, "utf-32"),
+                        (codecs.BOM_LE, "utf-16"),
+                        (codecs.BOM_BE, "utf-16"),
+                    ]:
+                        if byte_str.startswith(bytez):
+                            user_selected_encoding = confirm_encoding(guessed_encoding)
```

Comment:
IMO we don't need to ask the user here. If we have a BOM, we are pretty sure we guessed the right encoding and there's no reason to ask the user.